### ISSUE TITLE
Pyright can't infer types

### ISSUE THREAD

**Issue by foodornt** [microsoft/pyright#1933]
**Describe the bug**
I'm using Neovim Language Server protocol along with some plugins

I use nvim-lspconfig to initiate the lsp for python
Issuing this command: `lua vim.lsp.buf.hover()` shows the following prompt
![image](https://user-images.githubusercontent.com/40924367/120386274-19e23100-c331-11eb-849c-29ef3735a453.png)
Using discord.py library to create a discord bot
Installed visual studio code to try its intellisense out, it works just as expected, but pyright for some reason (maybe misconfig) isn't able to infer the types correctly

**To Reproduce**
1. Install neovim nightly `0.5.0`
2. Use a plugin manager and install `neovim/nvim-lspconfig` plugin
3. Setup the server as the following:
`require'lspconfig'.pyright.setup{ }`
4. `npm install -g pyright`
5. Try this snippet out (install discord.py using pip3)
```python
import discord
TOKEN_AUTH="YOUR TOKEN"

client = discord.Client()

@client.event
async def on_ready():
    guild = await client.fetch_guild(99999)
    me    = await guild.fetch_member(99999)
    me.voice

client.run(TOKEN_AUTH, bot=False)
```
6. Hover over `voice` in `me.voice`, type in the following command: `lua vim.lsp.buf.hover()`

**Expected behavior**
It is supposed (as it says in the picture) to show the type as `VoiceState`, but it shows `Unknown`

**VS Code extension or command-line**
I tried the [VS Code python extension](https://marketplace.visualstudio.com/items?itemName=ms-python.python) and as I said above it works very well.
Pyright + Neovim doesn't work as well.

**Comment by erictraut** [comment]
In general, type checkers like pyright require type stubs to declare the interfaces for each library. Do you you have type stubs for discord? In the absence of type stubs, all types imported from that library will be unknown.

Pyright does offer a mode that can be enabled through the "python.analysis.useLibraryCodeForTypes" setting. It will attempt to infer the types from the installed library code, assuming it is written in Python. Type inference can be error prone, so your mileage may vary with this approach. So if you can find good type stubs for the library, that's always preferable.

**Comment by foodornt** [comment]
that's EXACTLY the word I was looking for, type stubs :)

Thank you Eric 😆
